pkg/build/builder/sti.go

```diff
@@ -407,6 +410,6 @@ func copyToVolumeList(artifactsMapping []api.ImageSourcePath) (volumeList s2iapi
 	return
 }
 
-func convertS2IFailureType(reason s2iapi.StepFailureReason, message s2iapi.StepFailureMessage) (api.StatusReason, string) {
-	return api.StatusReason(reason), fmt.Sprintf("%s", message)
+func convertS2IFailureType(reason s2iapi.StepFailureReason, message s2iapi.StepFailureMessage) (api.BuildPhase, api.StatusReason, string) {
+	return api.BuildPhaseFailed, api.StatusReason(reason), fmt.Sprintf("%s", message)
```

Comment:
unless we really expect different s2i failure types to result in different build phases, I don't think you should do this... it's basically a function w/ a hardcoded return value which doesn't make much sense.  Just set the phase to failed explicitly in the places where you are calling this function.  I know it's an extra line of code but it seems like a better separation of concerns.